### ISSUE TITLE
Can't use column heading actions on calculated columns

### ISSUE THREAD

**Issue by mazameli** [metabase/metabase#8854]
![screen shot 2018-11-05 at 12 44 14 pm](https://user-images.githubusercontent.com/2223916/48025537-91f40600-e0f8-11e8-99a1-31353ef5a48e.png)

In RC 0.31, running from the jar, in the above scenario where my calculated column is defined as `Subtotal + (Tax * Total)`, if I click on any of the table heading actions nothing happens and I see JS errors like these. But aggregating calculated columns with the query builder itself seems to work just fine; the issue just appears to do with the column heading popover.

```
Uncaught Error: getFieldRefFromColumn expects non-null fieldId or column with non-null id
    at getFieldRefFromColumn (actions.js:72)
    at Object.question (app-main.bundle.js?15fac9025381a72d5834:68615)
    at ChartClickActions._this.handleClickAction (ChartClickActions.jsx:117)
    at onClick (ChartClickActions.jsx:253)
    at Object.invokeGuardedCallback (ReactErrorUtils.js:24)
    at executeDispatch (EventPluginUtils.js:83)
    at Object.executeDispatchesInOrder (EventPluginUtils.js:106)
    at executeDispatchesAndRelease (EventPluginHub.js:41)
    at executeDispatchesAndReleaseTopLevel (EventPluginHub.js:52)
    at Array.forEach (<anonymous>)
```
```
Uncaught Error: getFieldRefFromColumn expects non-null fieldId or column with non-null id
    at getFieldRefFromColumn (actions.js:72)
    at Object.question (app-main.bundle.js?15fac9025381a72d5834:68685)
    at ChartClickActions._this.handleClickAction (ChartClickActions.jsx:117)
    at onClick (ChartClickActions.jsx:253)
    at Object.invokeGuardedCallback (ReactErrorUtils.js:24)
    at executeDispatch (EventPluginUtils.js:83)
    at Object.executeDispatchesInOrder (EventPluginUtils.js:106)
    at executeDispatchesAndRelease (EventPluginHub.js:41)
    at executeDispatchesAndReleaseTopLevel (EventPluginHub.js:52)
    at Array.forEach (<anonymous>)
```

```
ReactErrorUtils.js:49 Uncaught Error: getFieldRefFromColumn expects non-null fieldId or column with non-null id
    at getFieldRefFromColumn (actions.js:72)
    at distribution (actions.js:245)
    at Question.distribution (Question.js:310)
    at Object.question (app-main.bundle.js?15fac9025381a72d5834:68737)
    at ChartClickActions._this.handleClickAction (ChartClickActions.jsx:117)
    at onClick (ChartClickActions.jsx:253)
    at Object.invokeGuardedCallback (ReactErrorUtils.js:24)
    at executeDispatch (EventPluginUtils.js:83)
    at Object.executeDispatchesInOrder (EventPluginUtils.js:106)
    at executeDispatchesAndRelease (EventPluginHub.js:41)
```

**Comment by nickhamlin** [comment]
Any chance this is related to https://github.com/metabase/metabase/issues/8940 ?

**Comment by camsaul** [comment]
Fixed by #8989

**Comment by RMMSecurity** [comment]
I checked out branches before and after some of Cam's and Saz's commits as well as master and I was able to reproduce it. 

Starting a new question using a saved question as the dataset. 

Check your console and you will see caught exceptions. Let's push on and try an action regardless: 

Group by a field and set it to a simple row count. Click on a cell to attempt to drill down and nothing happens visually. Check your console and it shows unhandled exceptions that were thrown. 

Both SQL as well as Custom QB were broken, but I've only tried my saved SQL questions this go around, but I can test!
